src/main/python/deep_qa/training/trainer.py

```diff
@@ -266,8 +285,13 @@ def train(self):
             debug_masks = self.debug_params.get('masks', [])
             debug_data = self.debug_params['data']
             if debug_data == "training":
-                self.debug_dataset = self.training_dataset
```

Comment:
`self._debug` looks at `self.debug_input`.  I think it'd be better here to just allow debugging on the _current_ dataset.  So, I would move this whole block to line 319, in the training dataset loop.  Then you don't need to use `training_[0-9]`.